contracts/Strategy.sol

```diff
@@ -53,80 +222,624 @@ contract Strategy is BaseStrategy {
             uint256 _debtPayment
         )
     {
-        // TODO: Do stuff here to free up any returns back into `want`
-        // NOTE: Return `_profit` which is value generated by all positions, priced in `want`
-        // NOTE: Should try to free up at least `_debtOutstanding` of underlying position
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            accrueInterest(i);
```

Comment:
example of possible refactoring
```
address _kashiPair = kashiPairs[i].pair;
uint _pid
accrueInterest(_kashiPair, _pid);
depositKashiInMasterChef(_kashiPair);
```

This avoids several reads to storage all the time, if the address is vetoed as part of the list the helper func should trust is correct

Reply:
Did this refactor, it resulted in substantial savings. 


Before
```
   ├─ constructor           -  avg: 5953382  avg (confirmed): 5953382  low: 5953382  high: 5953382
   ├─ harvest               -  avg: 1280166  avg (confirmed): 1280166  low:  831837  high: 2033843
   ├─ cloneKashiLender      -  avg:  817600  avg (confirmed):  817600  low:  817600  high:  817600
   ├─ adjustKashiPairRatios -  avg:  769088  avg (confirmed):  769088  low:  455927  high: 1177522
   ├─ removeKashiPair       -  avg:  399712  avg (confirmed):  399712  low:  399712  high:  399712
   ├─ tend                  -  avg:   55503  avg (confirmed):   55503  low:   55503  high:   55503
   ├─ addKashiPair          -  avg:   45370  avg (confirmed):   75296  low:   30732  high:   75296
   ├─ sweep                 -  avg:   37272  avg (confirmed):   54978  low:   27993  high:   54978
   ├─ setKeeper             -  avg:   29726  avg (confirmed):   29726  low:   29726  high:   29726
   ├─ initialize            -  avg:   29520  avg (confirmed):       0  low:   29082  high:   29958
   ├─ setStrategyName       -  avg:   29470  avg (confirmed):   29470  low:   14866  high:   44075
   └─ setEmergencyExit      -  avg:   24117  avg (confirmed):   24117  low:   24117  high:   24117```

After
```
   ├─ constructor           -  avg: 5972191  avg (confirmed): 5972191  low: 5972191  high: 5972191
   ├─ harvest               -  avg: 1181901  avg (confirmed): 1181901  low:  787393  high: 1862465
   ├─ cloneKashiLender      -  avg:  817667  avg (confirmed):  817667  low:  817667  high:  817667
   ├─ adjustKashiPairRatios -  avg:  669037  avg (confirmed):  669037  low:  296722  high: 1101352
   ├─ removeKashiPair       -  avg:  335453  avg (confirmed):  335453  low:  335453  high:  335453
   ├─ tend                  -  avg:   55503  avg (confirmed):   55503  low:   55503  high:   55503
   ├─ addKashiPair          -  avg:   45380  avg (confirmed):   75308  low:   30736  high:   75308
   ├─ sweep                 -  avg:   37282  avg (confirmed):   54994  low:   28001  high:   54994
   ├─ setKeeper             -  avg:   29730  avg (confirmed):   29730  low:   29730  high:   29730
   ├─ initialize            -  avg:   29520  avg (confirmed):       0  low:   29082  high:   29958
   ├─ setStrategyName       -  avg:   29470  avg (confirmed):   29470  low:   14866  high:   44075
   └─ setEmergencyExit      -  avg:   24117  avg (confirmed):   24117  low:   24117  high:   24117
```